File: Service/Billing/RMA.Service.Billing.Contracts/Entities/Refund.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;
using RMA.Service.Admin.SecurityManager.Contracts.Entities;

using System;
using System.Collections.Generic;

namespace RMA.Service.Billing.Contracts.Entities
{
    public class Refund
    {
        public decimal RefundAmount { get; set; }
        public DateTime RefundDate { get; set; }
        public Note Note { get; set; }
        public RefundTypeEnum? Trigger { get; set; }
        public RefundReasonEnum? RefundReason { get; set; }
        public List<int> TransactionIds { get; set; }
        public string RolePlayerName { get; set; }
        public int RolePlayerId { get; set; }
        public string FinPayeNumber { get; set; }
        public PeriodStatusEnum PeriodStatus { get; set; } = PeriodStatusEnum.Current;
        public List<RefundTransaction> PartialRefundTransactions { get; set; }
        public string GroupEmail { get; set; }
        public string ClientEmail { get; set; }
        public List<Role> requiredApprovalRoles { get; set; }
        public List<int> RolesThatHaveApproved { get; set; }
        public string TempDocumentKeyValue { get; set; }
        public int RolePlayerBankingId { get; set; }
        public bool OverrideMembershipApprover { get; set; }
        public List<RefundRmaBankAccountAmount> RefundRmaBankAccountAmounts { get; set; }
        public decimal? DebtorClaimRecoveryBalance { get; set; }
        public int? RefundHeaderId { get; set; } 
    }
}

File: Web/RMA-Web-Shell/projects/fincare/src/app/billing-manager/models/refund.ts
import { Note } from 'projects/shared-components-lib/src/lib/notes/note';
import { RolePlayerPolicy } from 'projects/clientcare/src/app/policy-manager/shared/entities/role-player-policy';
import { RefundTypeEnum } from '../../shared/enum/refund-type.enum';
import { RefundReasonEnum } from '../../shared/enum/refund-reason.enum';
import { PeriodStatusEnum } from 'projects/shared-models-lib/src/lib/enums/period-status-enum';
import { RefundTransaction } from './refund-transactions';
import { Role } from 'projects/shared-models-lib/src/lib/security/role';
import { BankAccountTypeEnum } from 'projects/shared-models-lib/src/lib/enums/bank-account-type-enum';
import { RolePlayerBankingDetail } from 'projects/shared-components-lib/src/lib/models/banking-details.model';
import { refundRmaBankAccountAmount } from './refundRmaBankAccountAmounts';

export class Refund {
  refundAmount: number;
  refundDate: Date;
  note: Note;
  trigger: RefundTypeEnum;
  refundReason: RefundReasonEnum;
  transactionIds: number[];
  rolePlayerId: number;
  rolePlayerName: string;
  requestCode: string;
  finPayeNumber: string;
  periodStatus: PeriodStatusEnum;
  partialRefundTransactions: RefundTransaction[];
  refundBankAccountNumber: string;
  refundBankBranchCode: string;
  groupEmail: string;
  clientEmail: string;
  tempDocumentKeyValue: string;
  rolePlayerBankingId: number;
  overrideMembershipApprover :boolean;
  refundRmaBankAccountAmounts: refundRmaBankAccountAmount[];
  debtorClaimRecoveryBalance: number;
}
